lib/api/core/validation/index.js

```diff
@@ -1,690 +1,680 @@
+   * @param {boolean} [verboseErrors]
+   * @returns {CollectionSpecification}
+   * @rejects InternalError
+   */
+  curateCollectionSpecification(indexName, collectionName, collectionSpec, dryRun = false, verboseErrors = false) {
+    let errorMessage = '';
+
+    return new Bluebird((resolve, reject) => {
+      const
+        treatedSpecification = {
+          strict: collectionSpec.strict || false,
+          fields: {},
+          validators: null
+        };
+
+      if (!checkAllowedProperties(collectionSpec, ['strict', 'fields', 'validators'])) {
+        errorMessage = `${indexName}.${collectionName}: the collection specification has invalid properties.`;
         if (verboseErrors) {
+          return resolve({isValid: false, errors: [errorMessage]});
+        }
+        return reject(new KuzzleInternalError(errorMessage));
+      }
+
+      if (collectionSpec.fields) {
+        try {
+          const result = this.structureCollectionValidation(collectionSpec, indexName, collectionName, verboseErrors);
           if (result.isValid === false) {
-            return resolve(result);
+            if (verboseErrors) {
+              // do not fail fast if we need verbose errors
+              return resolve(result);
+            }
+
+            const errorObject = new BadRequestError(result.errors[0]);
+            errorObject.details = result.errors;
+            return reject(errorObject);
           }
+          treatedSpecification.fields = result;
         }
-        return resolve({isValid: true});
-      })
-      .catch(error => {
-        // we do not want to reject since this method goal is to know what is going wrong with the given spec
-        return resolve({isValid: false, errors: [error]});
-      });
-  });
-};
-
-/**
- * @param {string} indexName
- * @param {string} collectionName
- * @param {CollectionSpecification} collectionSpec
- * @param {boolean} dryRun
- * @param {boolean} verboseErrors
- * @returns {CollectionSpecification}
- * @rejects InternalError
- */
-Validation.prototype.curateCollectionSpecification = function validationCurateCollectionSpecification (indexName, collectionName, collectionSpec, dryRun, verboseErrors) {
-  var
-    errorMessage = '',
-    result,
-    errorObject;
-
-  dryRun = dryRun || false;
-  verboseErrors = verboseErrors || false;
-
-  return new Promise((resolve, reject) => {
-    var
-      treatedSpecification = {
-        strict: collectionSpec.strict || false,
-        fields: {},
-        validators: null
-      };
+        catch (error) {
+          return reject(error);
+        }
+      }
 
-    if (!checkAllowedProperties(collectionSpec, ['strict', 'fields', 'validators'])) {
-      errorMessage = `${indexName}.${collectionName}: the collection specification has invalid properties.`;
-      if (verboseErrors) {
-        return resolve({isValid: false, errors: [errorMessage]});
+      if (collectionSpec.validators && Array.isArray(collectionSpec.validators)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 82 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)